controllers/resource/internal/vsphere.go

```diff
@@ -0,0 +1,43 @@
+package internal
+
+import (
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+func GetVSphereCredValues(credSecret *corev1.Secret) (map[string]string, error) {
```

Comment:
Is it possible, given the simplicity of this func, to encorporate this in the `reconciler_test.go` somehow so we can do away with the `test` pkg?

Reply:
This isn't a test function, it's used [here](https://github.com/aws/eks-anywhere/pull/3424/files#diff-42413ee5a1c110d438032dcbc28aa33f9af26f3235f367868883b4b5b3447579R158).

Reply:
Also, I see the confusion with my original comment 🤦🏻 . The above comment elaborates on my train of thought.

Reply:
Yeah, fair question. @g-gaston and I landed on this approach based on [this conversation](https://github.com/aws/eks-anywhere/pull/3424#discussion_r978872979).